### PR TITLE
Add cjdns protocol

### PR DESCRIPTION
**Pull request by ghost**
Needs testing though, will report back.

### CODE REVIEW

File: packages/wibed-battlemesh-experiment/files/etc/wbm/protos/cjdns.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+#ACTIONS are: add prepare clean?
+#example: bmx6.sh <action> <virtual_if> <actual_if> <IPv4/16> <IPv6/64>
+
+ACTION=$1
+LOGICAL_INTERFACE=$2
+REAL_INTERFACE=$3
+IPV4=$4
+IPV6=$5
+
+# TODO this action doesn't seem neccessary?
```
Comment (p4u) [inline packages/wibed-battlemesh-experiment/files/etc/wbm/protos/cjdns.sh]: Clean must clean the related configuration with the protocol. Including just the two first lines of the prepare() function should be enough. From prepare then you can call clean in the beginning.

**Comment by p4u** [comment]
Please, add cjdns as a dependency for the Makefile, thus it will be automatically selected. 
If cjdns is not available in the official OpenWRT feeds, you should perform one of both options:
1) Provide a feed and add it to the wibed repository
2) Push cjdns in the openwrt-routing official feed

**Comment by ghost** [comment]
> add cjdns as a dependency for the Makefile

Right, I remembered exactly this, and there is currently a problem with the dependencies of our packages [1]. The cjdns package depends on LuCI, since it makes use of `luci.model.uci` for conversion between UCI and cjdns' configuration format. I don't wanna make you pull in all of the LuCI stuff.

I'm working on resolving this right now, and tomorrow I'd be grateful if one of you could sit down with me, and walk me through a few things that are unclear to me at the moment.
1. The wibed-battlemesh-experiment package is not part of the images [2]. Do I build it locally and install it into the overlay?
2. What are the steps to get from "image flashed, with cjdns and all", to "being able to run an experiment"? Is creation of the logical interfaces automatic?
3. The protocol-specific scripts receive IP subnets/prefixes as arguments, but there's no way for cjdns to work with those. It'll assign its own address in fc00::/8, derived from the public key, as part of generating its config. IPv4 is not implemented at all. Might these two things cause problems? Any particular pieces of code I should read to get a better understanding of how the network gets configured?

[1] https://github.com/seattlemeshnet/meshbox
[2] http://wibed.confine-project.eu/wibed/last_trunk/ar71xx

**Comment by ghost** [comment]
> Any particular pieces of code I should read to get a better understanding of how the network gets configured?

Heh, probably the scripts `files/usr/bin/` :)

**Comment by p4u** [comment]
1. Right. In addition in the tar.gz you should also include all the ipk dependencies which are not in the base system (mainly the protocols). We can provide you such packet.
2. Image is not flashed, as you pointed out we will upload an overlay which will be removed at the end of the experiment
3. If you ignore the IPv6 prefix and use your own, should also work. However you would probably need to modify from your script the /etc/config/network file with the proper settings. IPv4 is a problem or not, depending on the experiments you want to run.

**Comment by ghost** [comment]
I'm getting there. The cjdns package needs one more update to remove the dependency on LuCI. I'll catch you later today (hopefully)!

**Comment by ghost** [comment]
I'll get back to this in time for WBMv8 :)

**Pull request closed without merging**